### PR TITLE
Release mistralai mistral 7 b instruct v0 1

### PR DESCRIPTION
**Pull request by choudharykartik**
Deployment Endpoint link: https://ml.azure.com/endpoints/realtime/mistral-7b-instruct-nc12/detail?wsid=/subscriptions/b17253fa-f327-42d6-9686-f3e553e24763/resourceGroups/donotuseordelete-model-validation-rg/providers/Microsoft.MachineLearningServices/workspaces/model-validations-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

model save to blob job link: https://ml.azure.com/runs/plucky_tiger_ftk5wgq6gs?wsid=/subscriptions/72c03bf3-4e69-41af-9532-dfcdc3eefef4/resourcegroups/shared-finetuning-rg/workspaces/v-swathikon&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

deployment response:
image

conda.yaml:
```
channels:
- conda-forge
dependencies:
- python=3.8.17
- pip<=23.2.1
- pip:
  - mlflow==2.6.0
  - cloudpickle==2.2.1
  - jsonpickle==3.0.2
  - mlflow-skinny==2.6.0
  - azureml-core==1.52.0
  - azureml-mlflow==1.52.0
  - azureml-metrics==0.0.27
  - scikit-learn==1.1.0
  - cryptography==41.0.2
  - python-dateutil==2.8.2
  - datasets==2.14.4
  - soundfile==0.12.1
  - librosa==0.10.0.post2
  - diffusers==0.19.3
  - sentencepiece==0.1.99
  - transformers==4.34.0
  - torch==2.0.0
  - accelerate==0.21.0
  - Pillow==9.4.0
  - azureml-evaluate-mlflow==0.0.27
- pycocotools=2.0.4
name: mlflow-env

```

MLmodel: 
```
flavors:
  hftransformersv2:
    code: code
    hf_predict_module: predict_mistral
    hf_config_class: AutoConfig
    hf_pretrained_class: MistralForCausalLM
    hf_tokenizer_class: AutoTokenizer
    huggingface_id: mistralai/Mistral-7B-Instruct-v0.1
    model_data: data
    pytorch_version: 2.0.0+cu117
    task_type: text-generation
    transformers_version: 4.34.0dev0
  python_function:
    code: code
    data: data
    env: conda.yaml
    loader_module: azureml.evaluate.mlflow.hftransformers
    python_version: 3.8.17
mlflow_version: 2.3.1
model_uuid: 3f4ac020ce904c628e3ac94591630a7a
utc_time_created: '2023-10-02 03:25:37.946855'

```

### CODE REVIEW

**Review by RSavage2 — APPROVED** [review]

File: assets/models/system/mistralai-Mistral-7B-Instruct-v0-1/description.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Task| Use case| Dataset| Python sample (Notebook)| CLI with YAML
+|--|--|--|--|--|
+Text generation | Text generation | <a href="https://huggingface.co/datasets/cnn_dailymail" target="_blank"> cnn_dailymail </a> | <a href="https://aka.ms/azureml-eval-sdk-text-generation/" target="_blank">evaluate-model-text-generation.ipynb</a> | <a href="https://aka.ms/azureml-eval-cli-text-generation/" target="_blank">evaluate-model-text-generation.yml</a>
+
+
+# **Inference samples**
+
+Inference type|Python sample (Notebook)|CLI with YAML
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-online-sdk-text-generation" target="_blank">text-generation-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-online-cli-text-generation" target="_blank">text-generation-online-endpoint.sh</a>
```
Comment (novaturient95) [inline assets/models/system/mistralai-Mistral-7B-Instruct-v0-1/description.md]: update to use dolly one

**Comment by novaturient95** [comment]
>    transformers_version: 4.34.0dev0

MLmodel has this while conda and requirements has 4.34.0. We would need to correct transformers version here as well. Let me do that

File: assets/models/system/mistralai-Mistral-7B-Instruct-v0-1/description.md (hunk 1 above)
Comment (choudharykartik) [reply assets/models/system/mistralai-Mistral-7B-Instruct-v0-1/description.md]: updated

File: assets/models/system/mistralai-Mistral-7B-Instruct-v0-1/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  inference-recommended-sku: Standard_NC12s_v3, Standard_NC24s_v3, Standard_NC24rs_v3,
+    Standard_ND40rs_v2, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
+  evaluation-recommended-sku: Standard_NC12s_v3, Standard_NC24s_v3, Standard_NC24rs_v3,
+    Standard_ND40rs_v2, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
+  languages: EN
+  azureml.copyRegistryFilesToWorkspace: true
+tags:
+  Featured: ''
+  Preview: ''
+  evaluation_compute_allow_list:
```
[the remark this answers is not in the document]
Comment (choudharykartik) [reply assets/models/system/mistralai-Mistral-7B-Instruct-v0-1/spec.yaml:15]: tested and added cpu computes

**Review by novaturient95 — APPROVED** [review]

**Review by RSavage2 — DISMISSED** [review]

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - Standard_E16s_v3
+  - Standard_NC12s_v3
+  - Standard_NC24s_v3
+  - Standard_NC24rs_v3
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  license: apache-2.0
+  task: text-generation
+  SharedComputeCapacityEnabled: ''
```
Comment (skasturi) [inline assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml:34]: author: mistralai

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+name: Mistral-7B-Instruct-v0-1
+path: ./
+properties:
+  SHA: 3dc28cf29d2edd31a0a7b8f0b21637059815b4d5
+  inference-recommended-sku: Standard_E16s_v3,Standard_NC12s_v3, Standard_NC24s_v3, Standard_NC24rs_v3,
+    Standard_ND40rs_v2, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
+  evaluation-recommended-sku: Standard_NC6s_v3,Standard_NC12s_v3, Standard_NC24s_v3, Standard_NC24rs_v3,
+    Standard_ND40rs_v2, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
+  languages: EN
+  azureml.copyRegistryFilesToWorkspace: true
```
Comment (skasturi) [inline assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml:11]: I don't think this is necessary for this model?

Comment (novaturient95) [reply assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml:11]: correct. Removed

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml (hunk 3 above)
Comment (novaturient95) [reply assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml]: added

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  Preview: ''
+  evaluation_compute_allow_list:
+  - Standard_NC6s_v3
+  - Standard_NC12s_v3
+  - Standard_NC24s_v3
+  - Standard_NC24rs_v3
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  inference_compute_allow_list:
```
[the remark this answers is not in the document]
Comment (novaturient95) [reply assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml:26]: done. Thanks

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: Mistral-7B-Instruct-v0-1
+path: ./
+properties:
+  SHA: 3dc28cf29d2edd31a0a7b8f0b21637059815b4d5
+  inference-recommended-sku: Standard_E16s_v3,Standard_NC12s_v3, Standard_NC24s_v3, Standard_NC24rs_v3,
+    Standard_ND40rs_v2, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
+  evaluation-recommended-sku: Standard_NC6s_v3,Standard_NC12s_v3, Standard_NC24s_v3, Standard_NC24rs_v3,
```
[the remark this answers is not in the document]
Comment (novaturient95) [reply assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml]: underneath these are also text-gen model so would support eval. But this is a similar problem to llama-chat.

For llama2-7b-chat, llama2-13b-chat we've evaluation compute allowlist. But not for llama-2-70b-chat

image

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: Mistral-7B-Instruct-v0-1
```
Comment (skasturi) [inline assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml:2]: Name needs to be "Mistral-7B-Instruct-v01"

Comment (novaturient95) [reply assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml]: updated

File: assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (novaturient95) [reply assets/models/system/Mistral-7B-Instruct-v0-1/spec.yaml]: removing

**Review by skasturi — APPROVED** [review]

**Pull request merged**